Modules/_operator.c

```diff
@@ -1595,36 +1595,59 @@ static PyType_Spec attrgetter_type_spec = {
 typedef struct {
     PyObject_HEAD
     PyObject *name;
-    PyObject *xargs; // reference to arguments passed in constructor
+    PyObject *args;
     PyObject *kwds;
     PyObject **vectorcall_args;  /* Borrowed references */
     PyObject *vectorcall_kwnames;
     vectorcallfunc vectorcall;
 } methodcallerobject;
 
-#ifndef Py_GIL_DISABLED
+
+#define _METHODCALLER_MAX_ARGS 8
+
+static PyObject *
+methodcaller_vectorcall(
+        methodcallerobject *mc, PyObject *const *args, size_t nargsf, PyObject* kwnames)
+{
+    if (!_PyArg_CheckPositional("methodcaller", PyVectorcall_NARGS(nargsf), 1, 1)
+        || !_PyArg_NoKwnames("methodcaller", kwnames)) {
+        return NULL;
+    }
+    assert(mc->vectorcall_args != NULL);
+
+    Py_ssize_t number_of_arguments = PyTuple_GET_SIZE(mc->args) +
+        (mc->vectorcall_kwnames? PyTuple_GET_SIZE(mc->vectorcall_kwnames):0);
+
+    PyObject *tmp_args[_METHODCALLER_MAX_ARGS];
+    tmp_args[0] = args[0];
+    assert(1 + number_of_arguments <= _METHODCALLER_MAX_ARGS);
+    memcpy(tmp_args + 1, mc->vectorcall_args, sizeof(PyObject *) * number_of_arguments);
+
+    PyObject *result = PyObject_VectorcallMethod(
+            mc->name, tmp_args,
+            (1 + PyTuple_GET_SIZE(mc->args)) | PY_VECTORCALL_ARGUMENTS_OFFSET,
+            mc->vectorcall_kwnames);
+
+    return result;
+}
+
 static int _methodcaller_initialize_vectorcall(methodcallerobject* mc)
 {
-    PyObject* args = mc->xargs;
+    PyObject* args = mc->args;
     PyObject* kwds = mc->kwds;
 
     Py_ssize_t nargs = PyTuple_GET_SIZE(args);
-    assert(nargs > 0);
     mc->vectorcall_args = PyMem_Calloc(
         nargs + (kwds ? PyDict_Size(kwds) : 0),
         sizeof(PyObject*));
     if (!mc->vectorcall_args) {
         PyErr_NoMemory();
         return -1;
     }
-    /* The first item of vectorcall_args will be filled with obj later */
-    if (nargs > 1) {
-        memcpy(mc->vectorcall_args, PySequence_Fast_ITEMS(args),
+    memcpy(mc->vectorcall_args, PySequence_Fast_ITEMS(args),
```

Comment:
I think we should prefer `_PyTuple_ITEMS` instead of `PySequence_Fast_ITEMS` for things that we know are tuples.